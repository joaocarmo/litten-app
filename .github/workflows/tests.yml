# This workflow will do a clean install of node dependencies, build the source
# code and run the available tests
name: Tests

on:
  push:
    branches: [ main, next ]
  pull_request:
    branches: [ main, next ]

env:
  CI: true
  FIREBASE_EMULATORS_PATH: $GITHUB_WORKSPACE/emulator-cache

jobs:
  test:
    environment: development
    runs-on: ubuntu-latest
    steps:
    - uses: actions/checkout@v3
    - name: Use Node.js v16 (LTS)
      uses: actions/setup-node@v3
      with:
        node-version: '16'
        cache: 'yarn'
    - name: Install firebase-tools globally
      run: yarn global add firebase-tools
    - name: Create .env file
      run: cp .env.example .env
    - name: Install dependencies
      run: yarn --frozen-lockfile
    - name: Lint the source files
      run: yarn lint
    - name: Type check
      run: yarn type-check
    - name: Validate the translations
      run: yarn translations-check
    - name: Execute the tests
      run: yarn test
    - name: Execute the firebase rules tests
      run: yarn test:firestore:rules
    - name: Execute the firebase model tests
      run: yarn test:firestore:model
    - name: Cache firebase emulators
      uses: actions/cache@v3
      with:
        path: ${{ env.FIREBASE_EMULATORS_PATH }}
        key: ${{ runner.os }}-firebase-emulators-${{ hashFiles('emulator-cache/*.{jar,zip}') }}
      continue-on-error: true
