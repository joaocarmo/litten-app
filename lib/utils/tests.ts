import { apps, clearFirestoreData } from '@firebase/rules-unit-testing'

export const projectId = 'litten-app'

export const cleanFirestore = async (): Promise<void> => {
  await Promise.all(apps().map((app) => app.delete()))
  await clearFirestoreData({
    projectId,
  })
}

export const mapName =
  <T>(key: string) =>
  (object: T): T =>
    Object.assign(object, {
      toString: function () {
        return this[key]
      },
    })

export const getRandInt = (min: num, max: num): num =>
  Math.floor(Math.random() * (max - min + 1) + min)
