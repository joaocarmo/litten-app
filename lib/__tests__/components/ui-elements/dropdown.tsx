import { render } from '@utils/tests/react-native'
import { UIDropdown } from '@ui-elements'

const messageOptions = [
  {
    key: 'all',
    label: 'All',
    value: 'all',
  },
  {
    key: 'read',
    label: 'Read',
    value: 'read',
  },
  {
    key: 'unread',
    label: 'Unread',
    value: 'unread',
  },
]

const handleOnSelect = jest.fn()

describe('Snapshot test for the "UIDropdown" component', () => {
  it('renders correctly', () => {
    const element = render(
      <UIDropdown
        onSelect={handleOnSelect}
        options={messageOptions}
        placement="bottom"
        selectedValue="all"
      />,
    )

    expect(element.toJSON()).toMatchSnapshot()
  })
})
