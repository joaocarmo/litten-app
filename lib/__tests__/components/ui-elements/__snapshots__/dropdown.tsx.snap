// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot test for the "UIDropdown" component renders correctly 1`] = `
<View
  style={
    [
      {
        "flexDirection": "row",
        "marginBottom": 6,
        "marginTop": 6,
      },
      undefined,
    ]
  }
>
  <Menu
    onSelect={[Function]}
    renderer={[Function]}
  >
    <MenuTrigger
      customStyles={
        {
          "TriggerTouchableComponent": {
            "$$typeof": Symbol(react.memo),
            "compare": null,
            "type": {
              "$$typeof": Symbol(react.forward_ref),
              "render": [Function],
            },
          },
        }
      }
    >
      <View
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#ffffff",
            "borderRadius": 20,
            "flexDirection": "row",
            "justifyContent": "space-between",
            "padding": 12,
          }
        }
      >
        <Text
          style={
            {
              "color": "#39a4f2",
              "fontWeight": "700",
            }
          }
        >
          All
        </Text>
        <SvgMock
          fill="#474545"
          height={12}
          style={
            {
              "transform": [
                {
                  "rotateZ": "90deg",
                },
              ],
            }
          }
          width={12}
        />
      </View>
    </MenuTrigger>
    <MenuOptions
      customStyles={
        {
          "optionsContainer": {
            "backgroundColor": "#ffffff",
            "borderColor": "#c4c4c4",
            "borderRadius": 6,
            "borderWidth": 0.5,
            "padding": 10,
            "shadowOffset": {
              "height": 0,
              "width": 0,
            },
            "shadowRadius": 0,
          },
        }
      }
    >
      <MenuOption
        customStyles={
          {
            "optionWrapper": {
              "paddingBottom": 10,
              "paddingRight": 64,
              "paddingTop": 10,
            },
          }
        }
        onSelect={[Function]}
        value="all"
      >
        <Text
          style={
            [
              {
                "color": "#6a6767",
                "fontWeight": "400",
                "padding": 2,
              },
              {
                "color": "#39a4f2",
              },
              undefined,
            ]
          }
        >
          All
        </Text>
      </MenuOption>
      <MenuOption
        customStyles={
          {
            "optionWrapper": {
              "paddingBottom": 10,
              "paddingRight": 64,
              "paddingTop": 10,
            },
          }
        }
        onSelect={[Function]}
        value="read"
      >
        <Text
          style={
            [
              {
                "color": "#6a6767",
                "fontWeight": "400",
                "padding": 2,
              },
              undefined,
              undefined,
            ]
          }
        >
          Read
        </Text>
      </MenuOption>
      <MenuOption
        customStyles={
          {
            "optionWrapper": {
              "paddingBottom": 10,
              "paddingRight": 64,
              "paddingTop": 10,
            },
          }
        }
        onSelect={[Function]}
        value="unread"
      >
        <Text
          style={
            [
              {
                "color": "#6a6767",
                "fontWeight": "400",
                "padding": 2,
              },
              undefined,
              undefined,
            ]
          }
        >
          Unread
        </Text>
      </MenuOption>
    </MenuOptions>
  </Menu>
</View>
`;
